### PR TITLE
Use GitHub as fallback for no/wrong URL in POM

### PR DESCRIPTION
**Pull request by zbynek**
Use GitHub's default readme for plugins that either don't define URL at all or reference plugins.jenkins.io as their doc URL (e.g. https://plugins.jenkins.io/rocketchatnotifier )

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

**Comment by daniel-beck** [comment]
How does this handle repos without a readme?

**Comment by zbynek** [comment]
Well you can also have plugins that explicitly set URL to github but don't have a readme. 

I'm not sure which services consume this, but the plugin site will display
"Documentation for this plugin is at github.com/plugin/project-without-readme"
and Update Center in Jenkins (in cases not covered by https://github.com/jenkinsci/jenkins/pull/4273 such as non-default update sites) should link to github.

Currently there are only 2 github URL that don't have a readme and will be affected by this PR:
https://github.com/jenkinsci/inline-pipeline-plugin
https://github.com/jenkinsci/groovy-events-listener-plugin-master (which seems to be misspelled https://github.com/jenkinsci/groovy-events-listener-plugin )
For 34 other plugins this will replace the "no documentation" message

**Comment by daniel-beck** [comment]
Thanks!

>  https://github.com/jenkinsci/inline-pipeline-plugin

Ooops. I only did that to have a test case for this 😆

**Review by timja — APPROVED** [review]

File: src/main/java/org/jvnet/hudson/update_center/Plugin.java [hunk 1]
```diff
@@ -166,12 +166,15 @@ public String getPluginUrl() throws IOException {
...
+        // last fallback: GitHub URL; also prevent plugins.j.io referencing itself
+        if (url == null || url.startsWith("https://plugins.jenkins.io")) {
+            url = requireTopLevelUrl(getScmUrl());
+        }
         String originalUrl = url;
 
         if (url != null) {
             url = url.replace("wiki.hudson-ci.org/display/HUDSON/", "wiki.jenkins-ci.org/display/JENKINS/");
-            url = url.replace("http://wiki.jenkins-ci.org", "https://wiki.jenkins-ci.org");
+            url = url.replace("http://wiki.jenkins-ci.org", "https://wiki.jenkins.io");
```
Comment (daniel-beck) [inline src/main/java/org/jvnet/hudson/update_center/Plugin.java]: This unrelated improvement made it unnecessarily cumbersome to review the impact of this change on the update center output. In the future, please separate out unnecessary changes.

**Comment by daniel-beck** [comment]
### Plugins that did not have documentation URLs and now point to the GitHub repo:

#### With standalone README:

* aliyun-oss-uploader: https://github.com/jenkinsci/aliyun-oss-uploader-plugin
* build-notifications: https://github.com/jenkinsci/build-notifications-plugin
* chef-cookbook-pipeline: https://github.com/jenkinsci/chef-cookbook-pipeline-plugin
* codebeamer-xunit-uploader: https://github.com/jenkinsci/codebeamer-xunit-uploader-plugin
* configuration-as-code-groovy: https://github.com/jenkinsci/configuration-as-code-groovy-plugin
* configuration-as-code-secret-ssm: https://github.com/jenkinsci/configuration-as-code-secret-ssm-plugin
* configuration-as-code-support: https://github.com/jenkinsci/configuration-as-code-plugin
* console-navigation: https://github.com/jenkinsci/console-navigation-plugin
* eiffel-broadcaster: https://github.com/jenkinsci/eiffel-broadcaster-plugin
* environment-variable-page-decoration: https://github.com/jenkinsci/environment-variable-page-decoration-plugin
* forensics-api: https://github.com/jenkinsci/forensics-api-plugin
* git-forensics: https://github.com/jenkinsci/git-forensics-plugin
* github-coverage-reporter: https://github.com/jenkinsci/github-coverage-reporter-plugin
* hashicorp-vault-pipeline: https://github.com/jenkinsci/hashicorp-vault-pipeline-plugin
* lambdatest-automation: https://github.com/jenkinsci/lambdatest-automation-plugin
* llvm-cov: https://github.com/jenkinsci/llvm-cov-plugin
* phabricator-k8s: https://github.com/jenkinsci/phabricator-k8s-plugin
* publish-over: https://github.com/jenkinsci/publish-over-plugin
* qualys-api-security: https://github.com/jenkinsci/qualys-api-security-plugin
* r7insight-log-forwarder: https://github.com/jenkinsci/r7insight-log-forwarder-plugin
* remoting-kafka: https://github.com/jenkinsci/remoting-kafka-plugin
* sonarqube-generic-coverage: https://github.com/jenkinsci/sonarqube-generic-coverage-plugin
* thycotic-vault: https://github.com/jenkinsci/thycotic-vault-plugin
* wavefront: https://github.com/jenkinsci/wavefront-plugin

#### Problematic READMEs

* build-token-trigger: https://github.com/jenkinsci/build-token-trigger-plugin (fairly useless README: link to plugin site with no more information)
* castecho: https://github.com/jenkinsci/castecho-plugin (useless README: dead wiki link)
* inline-pipeline: https://github.com/jenkinsci/inline-pipeline-plugin (no README)

### Formerly URLs pointed to plugin page on plugins.jenkins.io:

* https://plugins.jenkins.io/postbuildscript (has README)
* https://plugins.jenkins.io/prqa-plugin (useless README)
* https://plugins.jenkins.io/rhnpush-plugin (has README)
* https://plugins.jenkins.io/rocketchatnotifier (has README)
* https://plugins.jenkins.io/rpmsign-plugin (short README)

Overall a very nice change 👍

**Pull request merged**